build/bazel:  build chokes when `make` doesn't point to GNU make

On my system, `make` is a non-GNU make and the GNU make is called `gmake`.

When I run `gmake build` everything works, because GNU make propagates itself to sub-commands via the `MAKE` environment variable.

With `bazel build` I get the following errors:

```
ERROR: /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/c-deps/BUILD.bazel:10:15: CcConfigureMakeRule c-deps/libjemalloc/include failed: (Exit 1): bash failed: error executing command /usr/local/bin/bash -c ... (remaining 1 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox bash failed: error executing command /usr/local/bin/bash -c ... (remaining 1 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox

rules_foreign_cc: Build failed!
rules_foreign_cc: Keeping temp build directory  and dependencies directory  for debug.
rules_foreign_cc: Please note that the directories inside a sandbox are still cleaned unless you specify '--sandbox_debug' Bazel command line flag.

rules_foreign_cc: Printing build logs:

_____ BEGIN BUILD LOGS _____
Bazel external C/C++ Rules #0.0.8. Building library 'libjemalloc'
...
config.status: executing include/jemalloc/jemalloc.h commands
===============================================================================
jemalloc version   : 0.0.0-0-g0000000000000000000000000000000000000000
...
===============================================================================
make: "/tmp/tmp.D82ZsUSa/Makefile" line 3: Invalid line type
make: "/tmp/tmp.D82ZsUSa/Makefile" line 4: Invalid line type
make: "/tmp/tmp.D82ZsUSa/Makefile" line 72: Invalid line type
make: "/tmp/tmp.D82ZsUSa/Makefile" line 74: Invalid line type
make: "/tmp/tmp.D82ZsUSa/Makefile" line 75: Invalid line type
make: "/tmp/tmp.D82ZsUSa/Makefile" line 77: Invalid line type
make: "/tmp/tmp.D82ZsUSa/Makefile" line 79: Invalid line type
make: "/tmp/tmp.D82ZsUSa/Makefile" line 80: Invalid line type
```

There are a couple of problems:

- `bazel` picks up whatever `make` is found in PATH, probably needs to pin something instead

  ~This is not strictly a problem because it also gives me a workaround: I created a temporary `bin` directory with a symlink `make` -> `gmake` inside, and that solved the problem.~

- the command "make" is hardcoded inside the `c-deps/BUILD.bazel` scripts

I think the proper UX for this would be to have a version pin, plus an ability for the user to override e.g. via an env var, preferably `$MAKE`.

cc @rickystewart

Ah, but now I understand why you asked. here's the solution:

```
cmake(...
 cmake_options = ["-GUnix Makefiles"],
```

That works and brings me one step further.

Then I get a different kind of failure:

```
 cd . && /bin/sh /data/home/kena/.cache/bazel/_bazel_kena/9b9f53b7334495d6a871cbb472c466bb/sandbox/processwrapper-sandbox/1/execroot/cockroach/bazel-out/host/bin/external/rules_foreign_cc/toolchains/make.build_tmpdir/build-aux/missing automake-1.16 --foreign
configure.ac:35: error: version mismatch.  This is Automake 1.16.3,
configure.ac:35: but the definition used by this AM_INIT_AUTOMAKE
configure.ac:35: comes from Automake 1.16.1.  You should recreate
configure.ac:35: aclocal.m4 with aclocal and run automake again.
WARNING: 'automake-1.16' is probably too old.
         You should only need it if you modified 'Makefile.am' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'automake' program is part of the GNU Automake package:
         <https://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <https://www.gnu.org/software/autoconf>
         <https://www.gnu.org/software/m4/>
         <https://www.perl.org/>
make: *** [Makefile:1086: Makefile.in] Error 1
Target //pkg/cmd/cockroach-short:cockroach-short failed to build
```

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -166,11 +166,11 @@ c_deps()
 # aforementioned PRs.
 git_repository(
     name = "rules_foreign_cc",
-    commit = "6127817283221408069d4ae6765f2d8144f09b9f",
+    commit = "a3b0e5eaa723259458f5c85285f58e46ae7f25a2",
     remote = "https://github.com/cockroachdb/rules_foreign_cc",
 )
 
-load("@rules_foreign_cc//:workspace_definitions.bzl", "rules_foreign_cc_dependencies")
+load("@rules_foreign_cc//foreign_cc:repositories.bzl", "rules_foreign_cc_dependencies")
 
 rules_foreign_cc_dependencies()
 

--- a/c-deps/BUILD.bazel
+++ b/c-deps/BUILD.bazel
@@ -35,6 +35,12 @@ cmake(
         "CMAKE_BUILD_TYPE": "Release",
         "BUILD_LIBPROJ_SHARED": "OFF",
     },
+    # As of this writing (2021-05-05), foreign_cc
+    # only knows about windows, darwin and linux.
+    cmake_options = select({
+        "@io_bazel_rules_go//go/platform:windows": ["-GNinja"],
+        "//conditions:default": ["-GUnix Makefiles"],
+    }),
     lib_source = "@proj//:all",
     out_static_libs = ["libproj.a"],
     visibility = ["//visibility:public"],
@@ -48,6 +54,12 @@ cmake(
         "CMAKE_C_FLAGS": "-fPIC",
         "CMAKE_CXX_FLAGS": "-fPIC",
     },
+    # As of this writing (2021-05-05), foreign_cc
+    # only knows about windows, darwin and linux.
+    cmake_options = select({
+        "@io_bazel_rules_go//go/platform:windows": ["-GNinja"],
+        "//conditions:default": ["-GUnix Makefiles"],
+    }),
     lib_source = "@geos//:all",
     make_commands = [
         "mkdir -p libgeos/lib",
@@ -96,6 +108,12 @@ cmake(
         "CMAKE_TARGET_MESSAGES": "OFF",
         "CMAKE_BUILD_TYPE": "Release",
     },
+    # As of this writing (2021-05-05), foreign_cc
+    # only knows about windows, darwin and linux.
+    cmake_options = select({
+        "@io_bazel_rules_go//go/platform:windows": ["-GNinja"],
+        "//conditions:default": ["-GUnix Makefiles"],
+    }),
     lib_source = "@libroach//:all",
     make_commands = [
         "make roach",

--- a/c-deps/BUILD.bazel
+++ b/c-deps/BUILD.bazel
@@ -1,6 +1,6 @@
 # Load the components that lets us use cmake/make in third party deps.
-load("@rules_foreign_cc//tools/build_defs:cmake.bzl", "cmake_external")
-load("@rules_foreign_cc//tools/build_defs:configure.bzl", "configure_make")
+load("@rules_foreign_cc//foreign_cc:cmake.bzl", "cmake")
+load("@rules_foreign_cc//foreign_cc:configure.bzl", "configure_make")
 
 # TODO(irfansharif): All the cmake cache entries below were cargo-culted from
 # the Makefile. We may need to programatically set them depending on the
@@ -24,39 +24,24 @@ configure_make(
         "cp lib/libjemalloc.a libjemalloc/lib",
         "cp -r include libjemalloc",
     ],
-    static_libraries = ["libjemalloc.a"],
+    out_static_libs = ["libjemalloc.a"],
     visibility = ["//visibility:public"],
-)
-
-# Define the build targets for libprotobuf and protoc.
-cmake_external(
-    name = "libprotobuf",
-    binaries = ["protoc"],
-    cache_entries = {
-        "CMAKE_BUILD_TYPE": "Release",
-        "CMAKE_TARGET_MESSAGES": "OFF",
-        "protobuf_BUILD_TESTS": "OFF",
-    },
-    lib_source = "@protobuf//:all",
-    static_libraries = ["libprotobuf.a"],
-    visibility = ["//visibility:public"],
-    working_directory = "cmake",
 )
 
 # Define the build target for libproj.
-cmake_external(
+cmake(
     name = "libproj",
     cache_entries = {
         "CMAKE_BUILD_TYPE": "Release",
         "BUILD_LIBPROJ_SHARED": "OFF",
     },
     lib_source = "@proj//:all",
-    static_libraries = ["libproj.a"],
+    out_static_libs = ["libproj.a"],
     visibility = ["//visibility:public"],
 )
 
 # Define the targets for libgeos.
-cmake_external(
+cmake(
     name = "libgeos",
     cache_entries = {
         "CMAKE_BUILD_TYPE": "Release",
@@ -84,7 +69,7 @@ cmake_external(
             # TODO(#bazel): patchelf is also required here for release.
         ],
     }),
-    shared_libraries = select({
+    out_shared_libs = select({
         "@io_bazel_rules_go//go/platform:darwin": [
             "libgeos_c.dylib",
             "libgeos.dylib",
@@ -105,7 +90,7 @@ cmake_external(
 #
 # Bazel also expects the library archive and the include headers to be placed
 # in a certain path, so we fix it all up accordingly within make_commands.
-cmake_external(
+cmake(
     name = "libroach",
     cache_entries = {
         "CMAKE_TARGET_MESSAGES": "OFF",
@@ -118,7 +103,7 @@ cmake_external(
         "cp libroach.a libroach/lib/libroach.a",
         "cp -r $EXT_BUILD_ROOT/external/libroach/include libroach",
     ],
-    static_libraries = ["libroach.a"],
+    out_static_libs = ["libroach.a"],
     tools_deps = [
         "@libroach//:all",
     ],
@@ -150,6 +135,6 @@ configure_make(
         "mkdir -p libkrb5/lib",
         "cp libkrb5/libgssapi_krb5.a libkrb5/lib",
     ],
-    static_libraries = ["libgssapi_krb5.a"],
+    out_static_libs = ["libgssapi_krb5.a"],
     visibility = ["//visibility:public"],
 )

